src/Analyzer.Utilities/Extensions/ISymbolExtensions.cs

```diff
@@ -107,6 +107,68 @@ public static SymbolVisibility GetResultantVisibility(this ISymbol symbol)
 
             return visibility;
         }
+
+        public static bool MatchMemberDerived(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.ContainingType.DerivesFrom(type) && member.MetadataName == name;
+        }
+
+        public static bool MatchMethodDerived(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Method && member.MatchMemberDerived(type, name);
+        }
+
+        public static bool MatchMethod(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Method && member.MatchMember(type, name);
+        }
+
+        public static bool MatchPropertyDerived(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Property && member.MatchMemberDerived(type, name);
+        }
+
+        public static bool MatchFieldDerived(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Field && member.MatchMemberDerived(type, name);
+        }
+
+        public static bool MatchMember(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.ContainingType == type && member.MetadataName == name;
+        }
+
+        public static bool MatchProperty(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Property && member.MatchMember(type, name);
+        }
+
+        public static bool MatchField(this ISymbol member, INamedTypeSymbol type, string name)
```

Comment:
Consider renaming these `MatchAbc` methods to `MatchAbcByName` for clarity (for example, `MatchMethod` returns true for any method has the same name as the given symbol, it doesn't consider signature.)